src/components/create-notebook-dialog.tsx

```diff
@@ -394,7 +447,20 @@ export function CreateNotebookDialog({ children }: CreateNotebookDialogProps) {
           </div>
         )}
 
-        <div className="mt-6 flex justify-end">
+        <Label htmlFor="provider">Select Provider</Label>
+        <div className=" flex items-center justify-end gap-4">
+          <div className="w-full">
+            <Select
+              value={selectedProvider}
+              onChange={(e) => setSelectedProvider(e.target.value)}
+              options={[
```

Comment:
These should be mapped from a standalone list of providers. We can keep the API keys as secrets in k8s in prod. However we should allow the user to use their own keys.